### PR TITLE
Load list of `colonyOrVillage` from `PatientAddress`

### PR DESCRIPTION
**Pull request by Honey14**
https://app.clubhouse.io/simpledotorg/story/2572/load-list-of-colonyorvillage-from-the-patientaddress-table

### CODE REVIEW

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 1]
```diff
@@ -1256,7 +1256,6 @@ class PatientRepositoryAndroidTest {
   fun verify_deleted_prescribed_drugs_are_not_included_when_fetching_recent_patients() {
     val facilityUuid = currentFacility.uuid
     val recentPatient1 = savePatientWithPrescribedDrug(facilityUuid = facilityUuid)
-    val recentPatient2 = savePatientWithPrescribedDrug(facilityUuid = facilityUuid, deletedAt = Instant.now())
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1259]: Why was this removed? This is needed because we want to ensure that deleted patients are not fetched in the recent patients query.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 2]
```diff
@@ -1300,7 +1299,6 @@ class PatientRepositoryAndroidTest {
   fun verify_deleted_appointments_are_not_included_when_fetching_recent_patients() {
     val facilityUuid = UUID.fromString("086a5afd-c14b-4023-8eaa-9f29eee180a5")
     val recentPatient1 = savePatientWithAppointment(facilityUuid = facilityUuid, creationFacilityUuid = facilityUuid)
-    val recentPatient2 = savePatientWithAppointment(facilityUuid = facilityUuid, creationFacilityUuid = facilityUuid, deletedAt = Instant.now())
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1303]: Why was this removed? This is needed because we want to ensure that deleted patients are not fetched in the recent patients query.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 3]
```diff
@@ -1317,23 +1315,28 @@ class PatientRepositoryAndroidTest {
...
-    val recentPatient2 = savePatientWithAppointment(
-        facilityUuid = facilityUuid,
-        creationFacilityUuid = facilityUuid,
-        status = Cancelled
-    )
-    val recentPatient3 = savePatientWithAppointment(
-        facilityUuid = facilityUuid,
-        creationFacilityUuid = facilityUuid,
-        status = Visited
-    )
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1329]: These are needed because we want to verify that only the right patients are fetched in the recent patients query.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 4]
```diff
@@ -1354,10 +1357,7 @@ class PatientRepositoryAndroidTest {
   @Test
   fun verify_only_patients_with_manual_appointments_are_included_when_fetching_recent_patients() {
     val facilityUuid = UUID.fromString("e86e036c-8a55-4a07-8678-32f13b006dcb")
-    val recentPatient1 = savePatientWithAppointment(facilityUuid = facilityUuid, creationFacilityUuid = facilityUuid, appointmentType = Automatic)
     val recentPatient2 = savePatientWithAppointment(facilityUuid = facilityUuid, creationFacilityUuid = facilityUuid, appointmentType = Manual)
-    val recentPatient3 = savePatientWithAppointment(facilityUuid = facilityUuid, creationFacilityUuid = facilityUuid,
-        appointmentType = AppointmentType.Unknown(""))
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1360]: These should not have been deleted, for similar reasons as mentioned earlier.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 5]
```diff
@@ -1317,23 +1315,28 @@ class PatientRepositoryAndroidTest {
...
+    val address3 = testData.patientAddress(colonyOrVillage = "138, residency road")
+
+    val addresses = listOf(address1, address2, address3)
+    database.addressDao().save(addresses)
+
+    val expectedListOfColoniesOrVillages = listOf(address1.colonyOrVillage, address2.colonyOrVillage, address3.colonyOrVillage)
+
+    val listOfColoniesOrVillages = patientRepository.allColoniesOrVillagesInPatientAddress()
+    assertThat(listOfColoniesOrVillages).isEqualTo(expectedListOfColoniesOrVillages)
+  }
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1338]: Hmm, are we populating this from the addresses that already exist? I was under the impression that we will be fetching a list from the server in order to encourage a set of known names. Was this changed?

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 6]
```diff
@@ -1317,23 +1315,28 @@ class PatientRepositoryAndroidTest {
...
         .recentPatients(facilityUuid)
         .blockingFirst()
     assertThat(recentPatients).isEqualTo(listOf(recentPatient1))
   }
 
+  @Test
+  fun verify_only_list_of_colonies_or_villages_are_received_from_patient_address() {
+    val address1 = testData.patientAddress(colonyOrVillage = "Sharma Courts")
+    val address2 = testData.patientAddress(colonyOrVillage = "7111 Jain Mountains")
+    val address3 = testData.patientAddress(colonyOrVillage = "138, residency road")
```
Comment (vinaysshenoy) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1329]: We should ideally also create duplicate addresses and ensure that final set of results are distinct and have no repeating elements.

File: app/src/main/java/org/simple/clinic/patient/PatientAddress.kt [hunk 7]
```diff
@@ -98,5 +98,8 @@ data class PatientAddress(
 
     @Query("SELECT COUNT(uuid) FROM PatientAddress")
     abstract fun count(): Int
+
+    @Query("SELECT DISTINCT colonyOrVillage FROM PatientAddress")
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/patient/PatientAddress.kt:102]: Do we need to order them in any specific way? I'd assume alphabetical ordering is preferred, otherwise it would be hard to find a location from a long list of names.

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt (hunk 5 above)
Comment (Honey14) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:1338]: We're using the current sync region’s patient addresses to populate the village names. Here's the updated conversation : https://simpledotorg.slack.com/archives/C01FN9GCNAJ/p1611753504014400?thread_ts=1611542193.001000&cid=C01FN9GCNAJ

**Review by vinaysshenoy — APPROVED** [review]
